controllers/openstackdataplane_controller.go

```diff
@@ -50,10 +60,146 @@ type OpenStackDataPlaneReconciler struct {
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
+		// helper might be nil, so can't use util.LogErrorForObject since it requires helper as first arg
+		r.Log.Error(err, fmt.Sprintf("unable to acquire helper for OpenStackDataPlane %s", instance.Name))
+		return ctrl.Result{}, err
+	}
+
+	// Always patch the instance status when exiting this function so we can persist any changes.
+	defer func() {
+		// update the overall status condition if service is ready
+		if instance.IsReady() {
+			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, condition.ReadyMessage)
+		} else {
+			// something is not ready so reset the Ready condition
+			instance.Status.Conditions.MarkUnknown(
+				condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
+			// and recalculate it based on the state of the rest of the conditions
+			instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.ReadyCondition))
+		}
+
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			r.Log.Error(_err, "PatchInstance error")
+			_err = err
+			return
+		}
+	}()
+
+	if instance.Status.Conditions == nil {
+		instance.InitConditions()
+		// Register overall status immediately to have an early feedback e.g. in the cli
+		return ctrl.Result{}, nil
+	}
+
+	// Reset all ReadyConditons to 'Unknown'
+	instance.InitConditions()
+
+	ctrlResult, err := CreateDataPlaneResources(ctx, instance, helper)
+	if err != nil {
+		return ctrl.Result{}, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	var deployErrors []string
+	shouldRequeue := false
+	if instance.Spec.DeployStrategy.Deploy {
+		r.Log.Info("Starting DataPlane deploy")
+		r.Log.Info("Set ReadyCondition false")
+		roles := &dataplanev1beta1.OpenStackDataPlaneRoleList{}
+
+		listOpts := []client.ListOption{
+			client.InNamespace(instance.GetNamespace()),
+		}
+		labelSelector := map[string]string{
+			"openstackdataplane": instance.Name,
+		}
+		if len(labelSelector) > 0 {
+			labels := client.MatchingLabels(labelSelector)
+			listOpts = append(listOpts, labels)
+		}
+		err = helper.GetClient().List(ctx, roles, listOpts...)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		instance.Status.Conditions.Set(condition.FalseCondition(condition.ReadyCondition, condition.InitReason, condition.SeverityInfo, dataplanev1beta1.DataPlaneRoleReadyWaitingMessage))
+		for _, role := range roles.Items {
+			if role.Spec.DataPlane != instance.Name {
+				err = fmt.Errorf("role %s: role.DataPlane does not match with role.Label", role.Name)
+				deployErrors = append(deployErrors, "role.Name: "+role.Name+" error: "+err.Error())
+			}
+			role.Spec.DeployStrategy.Deploy = instance.Spec.DeployStrategy.Deploy
```

Comment:
You need to move this line inside the CreateOrPatch func, otherwise it doesn't take effect.